src/bundle/ui-dev/src/modules/universal-discovery/search.tab.module.js

```diff
@@ -1,20 +1,33 @@
-import React, { useContext } from 'react';
+import React, { useContext, useEffect, useRef } from 'react';
 
 import Tab from './components/tab/tab';
 import Search from './components/search/search';
 
-import { TabsConfigContext } from './universal.discovery.module';
+import { LoadedLocationsMapContext, MarkedLocationIdContext, TabsConfigContext } from './universal.discovery.module';
 
 const { ibexa, Translator } = window;
 
 const SearchTabModule = () => {
     const tabsConfig = useContext(TabsConfigContext);
+    const shouldRestorePreviousStateRef = useRef(true);
+    const [markedLocationId, setMarkedLocationId] = useContext(MarkedLocationIdContext);
+    const [loadedLocationsMap, dispatchLoadedLocationsAction] = useContext(LoadedLocationsMapContext);
+
     const actionsDisabledMap = {
         'content-create-button': false,
         'sort-switcher': true,
         'view-switcher': true,
     };
 
+    useEffect(() => {
+        return () => {
+            if (shouldRestorePreviousStateRef.current) {
```

Comment:
What's the point of this check with current state of this ref? I see it's been initialized on true, but it's not changed later, is it will always be true